systemservices/marketplace/mesg.yml

```diff
@@ -72,17 +72,99 @@ tasks:
                                 type: Object
                                 repeated: true
                                 optional: true
+                                object:
+                                  key:
+                                    type: String
+                                  name:
+                                    type: String
+                                    optional: true
+                                  description:
+                                    type: String
+                                    optional: true
+                                  inputs: &manifestServiceDataParameters
+                                    type: Object
+                                    optional: true
+                                    repeated: true
+                                    object: 
+                                      key:
+                                        type: String
+                                      name:
+                                        type: String
+                                        optional: true
+                                      description:
+                                        type: String
+                                        optional: true
+                                      type:
+                                        type: String
+                                      optional:
+                                        type: Boolean
+                                        optional: true
+                                      repeated:
+                                        type: Boolean
+                                        optional: true
+                                      object:
+                                        type: Object
```

Comment:
we could have typed Parameter here as well if we implemented this feature https://forum.mesg.com/t/service-definition-support-recursive-types/260/6